### PR TITLE
Drop python 3.7

### PR DESCRIPTION
**Pull request by tommy-waltmann**
## Description

This PR removes support for python 3.7 by dropping it from our testing matrix and build wheels operations.

## Motivation and Context

Python 3.7 hit end of life in June: https://devguide.python.org/versions/

## How Has This Been Tested?

If the current tests pass, we are ready to do.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: .github/workflows/cibuildwheel-before-test.sh [hunk 1]
```diff
@@ -6,10 +6,10 @@ fi
 
 PACKAGE_DIR=$1
 
-if [[ $(python --version 2>&1) == *"3.7."* ]]; then
+if [[ $(python --version 2>&1) == *"3.8."* ]]; then
   # Python 3.7 is only supported with oldest requirements
```
Comment (b-butler) [inline .github/workflows/cibuildwheel-before-test.sh]:
```suggestion
  # Python 3.8 is only supported with oldest requirements
```

File: .github/workflows/cibuildwheel-before-test.sh [hunk 2]
```diff
@@ -6,10 +6,10 @@ fi
...
   pip install -U -r "${PACKAGE_DIR}/.github/workflows/oldest-test-reqs.txt" --progress-bar=off
-elif [[ $(python --version 2>&1) == *"3.8."* ]]; then
+elif [[ $(python --version 2>&1) == *"3.9."* ]]; then
   # Python 3.8 needs compatible requirements
```
Comment (b-butler) [inline .github/workflows/cibuildwheel-before-test.sh]:
```suggestion
  # Python 3.9 needs compatible requirements
```

**Review by b-butler — CHANGES_REQUESTED** [review]
Some comments need to be updated.

**Comment by tommy-waltmann** [comment]
@b-butler would appreciate a review here, this PR is holding up all the new dependabot PRs.

**Pull request merged**